docs/deployment/k8s.md

```diff
@@ -0,0 +1,519 @@
+            - secretRef:
+                name: backstage-secrets
+```
+
+For production deployments, the `image` reference will usually be a full URL to
+a repository on a container registry (for example, ECR on AWS).
+
+For testing locally with `minikube`, you can point the local Docker daemon to
+the `minikube` internal Docker registry and then rebuild the image to install
+it:
+
+```shell
+$ eval $(minikube docker-env)
+$ yarn build-image --tag backstage:1.0.0
+```
+
+There is no special wiring needed to access the PostgreSQL service. Since it's
+running on the same cluster, Kubernetes will inject `POSTGRES_SERVICE_HOST` and
+`POSTGRES_SERVICE_PORT` environment variables into our Backstage container.
+These can be used in the Backstage `app-config.yaml` along with the secrets:
+
+```yaml
+backend:
+  database:
+    client: pg
+    connection:
+      host: ${POSTGRES_SERVICE_HOST}
+      port: ${POSTGRES_SERVICE_PORT}
+      user: ${POSTGRES_USER}
+      password: ${POSTGRES_PASSWORD}
+```
+
+Make sure to rebuild the Docker image after applying `app-config.yaml` changes.
+
+Apply this Deployment to the Kubernetes cluster:
+
+```shell
+$ kubectl apply -f kubernetes/backstage.yaml
+deployment.apps/backstage created
+
+$ kubectl get deployments --namespace=backstage
+NAME        READY   UP-TO-DATE   AVAILABLE   AGE
+backstage   1/1     1            1           1m
+postgres    1/1     1            1           10m
+
+$ kubectl get pods --namespace=backstage
+NAME                                 READY   STATUS    RESTARTS   AGE
+backstage-54bfcd6476-n2jkm           1/1     Running   0          58s
+postgres-56c86b8bbc-66pt2            1/1     Running   0          9m
+```
+
+Beautiful! 🎉 The deployment and pod are running in the cluster. If you run into
+any trouble, check the container logs from the pod:
+
+```shell
+# -f to tail, <pod> -c <container>
+$ kubectl logs --namespace=backstage -f backstage-54bfcd6476-n2jkm -c backstage
+```
+
+### Creating a Backstage service
+
+Like the [PostgreSQL service](#creating-a-postgresql-service) above, we need to
+create a Kubernetes Service for Backstage to handle connecting requests to the
+correct pods.
+
+Create the Kubernetes Service descriptor:
+
+```yaml
+# kubernetes/backstage-service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: backstage
+  namespace: backstage
+spec:
+  selector:
+    app: backstage
+  ports:
+    - name: http
+      port: 80
+      targetPort: http
+```
+
+The `selector` here is telling the Service which pods to target, and the port
+mapping translates normal HTTP port 80 to the backend http port (7000) on the
+pod.
+
+Apply this Service to the Kubernetes cluster:
+
+```shell
+$ kubectl apply -f kubernetes/backstage-service.yaml
+service/backstage created
+```
+
+Now we have a fully operational Backstage deployment! 🎉 For a _**grand
+reveal**_, you can forward a local port to the service:
+
+```shell
+$ sudo kubectl port-forward --namespace=backstage svc/backstage 80:80
```

Comment:
```suggestion
$ sudo kubectl port-forward --namespace=backstage service/backstage 80:80
```

isn't this what it should be? i actually don't know these things

Reply:
[Surprisingly not](https://github.com/kubernetes/kubernetes/pull/59809), I would have thought the same since k8s doesn't usually abbreviate things.